src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -178,10 +179,26 @@
     /** Period literal. */
     private static final String PERIOD = ".";
 
+    /** Summary tag text. */
+    private static final String SUMMARY_TEXT = "@summary";
+
     /** Set of allowed Tokens tags in summary java doc. */
     private static final Set<Integer> ALLOWED_TYPES = Collections.unmodifiableSet(
             new HashSet<>(Arrays.asList(JavadocTokenTypes.TEXT,
-                    JavadocTokenTypes.WS))
+                    JavadocTokenTypes.WS,
+                    JavadocTokenTypes.DESCRIPTION,
+                    JavadocTokenTypes.TEXT))
+    );
+
+    /**
+     * Set of allowed Tokens tags in summary java doc.
+     */
+    private static final Set<String> ALLOWED_INLINE_TAGS = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    "@code",
+                    "@link",
+                    "@input"
```

Comment:
please explain why such tags are allowed?

Reply:
Sir, I could not find any proper documentation which specifies which tags are allowed inside Inline Tag, but with respect to https://docs.oracle.com/javase/10/docs/specs/doc-comment-spec.html#general-syntax, it stats that 
 **_The first sentence of the initial description should be a summary sentence that contains a concise but complete description of the declared entity. Descriptive text may include HTML tags and entities, and [inline tags](https://docs.oracle.com/javase/10/docs/specs/doc-comment-spec.html#inline-tags) as described below._**

I also, had added all the tags (which were not present earlier ) that this documentation has specified except `@inheritDoc` as it is forbidden.

Reply:
please move this explanation to documentation of Check as users need to know it.

Reply:
Done.

Reply:
good, but meant html documentation of Check.
https://checkstyle.org/config_javadoc.html#SummaryJavadoc

you do very big update for Check
please add few words to description, please add example of summary tag to examples.
We need add link https://docs.oracle.com/javase/10/docs/specs/doc-comment-spec.html#general-syntax to description to educate users. it is good to place 
`The first sentence of the initial description should be a summary sentence that contains a concise but complete description of the declared entity. Descriptive text may include HTML tags and entities, and inline tags as described below.`  as quote to description because there are a lot of text in this documentation and people would unlikely to read all, so we give them main part.

Reply:
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/FixNewSummaryFormat_2021213939/config_javadoc.html#SummaryJavadoc

Please let me know if this works or not. I had updated the documentation.